app/src/main/java/org/stepic/droid/ui/util/PopupHelper.kt

```diff
@@ -56,16 +60,28 @@ object PopupHelper {
         popupArrowView.setBackgroundResource(theme.arrowRes)
         popupArrowView.changeVisibility(withArrow)
 
-        if (withArrow) {
-            popupView.viewTreeObserver.addOnGlobalLayoutListener {
-                popupArrowView.x = calcArrowHorizontalOffset(anchorView, popupView, popupView.arrowView)
-            }
-        }
-
         val popupWindow = PopupWindow(popupView, LinearLayout.LayoutParams.WRAP_CONTENT, LinearLayout.LayoutParams.WRAP_CONTENT)
         popupWindow.animationStyle = R.style.PopupAnimations
         popupWindow.isOutsideTouchable = cancelableOnTouchOutside
 
+        if (withArrow) {
+            popupView.viewTreeObserver.addOnGlobalLayoutListener(object : ViewTreeObserver.OnGlobalLayoutListener {
+                override fun onGlobalLayout() {
+                    popupArrowView.x = calcArrowHorizontalOffset(anchorView, popupView, popupView.arrowView)
+                    if (isAboveAnchor) {
```

Comment:
Sometimes popup displayed truncated, maybe there is an another way to measure y offset.